addons/mail/models/models.py

```diff
@@ -511,6 +511,7 @@ def email_key(email):
             ))
 
             recipients = [{
+                'display_name': partner.display_name,
```

Comment:
And this one would need to be optimizing manually by appending only when necessary.

Reply:
It's not useless noise, it's important information.

Users that want to add the invoice partner do not know/do not care about the email itself.

Unnamed is buggy and confusing.
Invoice should be displayed in that case.

<img width="910" height="481" alt="image" src="https://github.com/user-attachments/assets/252dc928-ab98-4868-84a2-1cea1d71c882" />